### PR TITLE
gitlab: add gitlab-pages

### PR DESCRIPTION
**Pull request by hectorj2f**
Fixes:

Adding support for gitlab-pages which uses gitlab-cng-base packages.

Related: 
https://gitlab.com/gitlab-org/gitlab-pages

https://github.com/chainguard-dev/image-requests/issues/462

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For version bump PRs

- [x] The `epoch` field is reset to 0

### CODE REVIEW

File: gitlab-pages.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package:
```
Comment (rawlingsj) [inline gitlab-pages.yaml:1]:
```suggestion
# Source is on gitlab so we can't use github for updates
#nolint:git-checkout-must-use-github-updates
```

Comment (rawlingsj) [reply gitlab-pages.yaml:1]: this will silence the lint failure

**Review by imjasonh — DISMISSED** [review]

File: gitlab-pages.yaml (hunk 1 above)
Comment (hectorj2f) [reply gitlab-pages.yaml:3]: I added `package:` back which was missing.

File: gitlab-pages.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - license: MIT
+  dependencies:
+    runtime:
+      - gitlab-cng-base
+      - gitlab-cng-pages-scripts
+
+environment:
+  contents:
+    packages:
+      - gitlab-cng-base
```
Comment (mattmoor) [inline gitlab-pages.yaml:20]: Is this really a build time dep?

File: gitlab-pages.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  # The release-monitor identified is for the gitlab-runner,
+  # but these seem to be co-versioned.
+  version: 16.4.1
+  epoch: 0
+  description: GitLab Pages daemon used to serve static websites for GitLab users.
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - gitlab-cng-base
```
Comment (mattmoor) [inline gitlab-pages.yaml:14]: This doesn’t exist until your other PR lands?

File: gitlab-pages.yaml (hunk 2 above)
Comment (hectorj2f) [reply gitlab-pages.yaml:20]: Adding busybox as dependency.

**Review by imjasonh — APPROVED** [review]

**Pull request merged**